### DISCUSSION TITLE
Brainstorming Funktionen Forum

### DISCUSSION THREAD

**Discussion by Cosnavel** [Webmasters-Europe/codemeetup-forum#11] [Ideas]
.

**Comment by milack68** [comment]
Ich habe mir einmal auf die Schnelle Gedanken zu den grundlegenden Funktionen gemacht:

Forum => Thema => Post => PostReply
**Beispiele** 
Webentwicklung => PHP für Einsteiger => durch Arrays iterieren => Vielen Dank, ich glaube so könnte man es auch machem .....
Webentwicklung => CSS => Dropdown-Menü mit TailwindCSS stylen

**Benutzerberechtigungen / Rollen**
- [ ] Admin
- [ ] User

**Admin-Funktionen**
- [ ] create roles and permissions
- [ ] confirm user
- [ ] add special role to user
- [ ] create forum

### User-Funktionen
- [ ] registration
- [ ] login
- [ ] read forum
- [ ] create, read theme
- [ ] create, read, update, delete (crud) post (update and delete only 5 minutes after creating and no replys to the post)
- [ ] crud postreply
- [ ] subscribe to forum
- [ ] subscribe to theme
- [ ] get message when another user replys to your post (switch on or off)
- [ ] update profile
- [ ] send message to other user
- [ ] upload files

**Reply by Cosnavel** [reply]
Klingt gut. Mich interessiert die Meinung der Anderen dazu. @milack68 kannst du für das Auth schon einmal Feature Requests anlegen?

**Comment by ghost** [comment]
### Wow, Michael. Da ist ja schon sehr viel dabei. Hast du schon Erfahrungen damit?
#### Hier noch ein paar Ideen:

* Benutzerberechtigungen / Rollen: 
  * wie wäre es mit einer Art Gastrechten/Gastaccount, wenn sich jemand erstmal nur im Forum umschauen möchte um herauszufinden, ob das etwas für ihn ist?
* User-Funktionen
  * eine einfache Art den eigenen Account zu löschen/unsubscribe o.ä. 

* Was haltet ihr von einer News-Section o.ä.?

**Reply by milack68** [reply]
@Klingohle Gute Idee mit der News-Section.
Zu den Gastrechten hätte ich grundsätzlich gedacht, dass das Forum öffentlich lesbar ist. Erst wenn man posten will muss man sich registrieren. Wir können aber auch ein geschlossenes Forum machen, dann ist aber die Frage, ob man hier Gästeaccounts möchte, weil das Forum wäre ja dann nicht ohne Grund geschlossen.

Erfahrungen diesbezüglich habe ich nur mit der Software phpBB. Ich war früher einige Jahre Präsident der Rechtspflegervereinigung - ein Verein, dem zirka 800 Mitglieder (= Rechtspfleger) angehören und der u.a. ein österreichweites Diskussionsforum betreibt. Ich habe das damals eingeführt und administriere es heute noch. Das ist ein geschlossenes Forum - man muss sich nicht nur registrieren sondern es muss zusätzlich noch eine Freischaltung durch einen Admin erfolgen (dieser prüft im Wesentlichen, ob der User ein Vereinsmitglied ist). Hier wäre zB ein Gastzugang nicht gewünscht, weil man verhindern will, dass bestimmte Diskussionsthemen öffentlich werden.

**Reply by ghost** [reply]
#### Wenn das Forum ganz offen ist, braucht man keine Gastrechte o.ä. - wäre ok.
#### Beeindruckend, was du so an Erfahrung hast. Da kannst du  in dem Projekt sicherliche gute Erfahrungswerte beisteuern. Ich bin wirklich gespannt, was wir da hinbekommen in den nächsten Wochen :-)

**Reply by thelf** [reply]
Vielen Dank Michael für den umfangreichen Aufbau, was mir noch einfallen würde als Ergänzung

**Generell für das Forum:**
- Eine Such Funktion
- Eine Anzeige der Anzahl der Beiträge zu einem Thema
- Eine Anzeige der Anzahl der Beiträge und Themen generell

**User-Funktionen:** 
- Passwort vergessen Funktion

**Admin-Funktionen:** 
- Benutzer sperren/entfernen

@Klingohle Ich finde das mit der News-Section eine gute Idee, ist damit die Anzeige der letzten Beiträge aus dem Forum gemeint oder dachtest du an andere News?

**Reply by ghost** [reply]
Hi, die News-Section kann alles mögliche umfassen, klar, auch letzte Einträge,. Aus meiner Sicht wären aber auch Verweise auf Events wie das nächste Code Meetup ;-) oder auch Hinweise auf hilfreiche Informationsquellen wie der Link von Alena  sinnvoll. Mein Tipp wäre, das variabel zu halten.

**Comment by alenavershinina** [comment]
Michael hat alles super strukturiert. Ich glaube, waere es noch bequem fuer User, nur seine letzte Eintraege zu sehen (in seinem Profil oder News-Section).
Vielleicht brauchen wir noch eine Seite mit Forum-Regeln (oder sie koennen gleich in eine Thema in Forum sein).
